src/test/groovy/org/arl/fjage/test/spec/fjageSpec.js

```diff
@@ -414,6 +415,117 @@ describe('Shell GetFile/PutFile', function () {
       });
     });
   });
+
+  it('should be able to append a file using PutFileReq', function (done) {
+    const pfr = new PutFileReq();
+    pfr.recipient = shell;
+    pfr.filename = DIRNAME + '/' + FILENAME;
+    pfr.contents = Array.from((new TextEncoder('utf-8').encode(TEST_STRING + ' ' + TEST_STRING)));
+    const rsp = gw.request(pfr);
+    expect(rsp).not.toBeNull();
+    rsp.then(msg => {
+      expect(msg).toBeTruthy();
+      expect(msg.perf).toEqual(Performative.AGREE);
+      var gfr = new GetFileReq();
+      gfr.recipient = shell;
+      gfr.filename = DIRNAME + '/' + FILENAME;
+      const rsp2 = gw.request(gfr);
+      expect(rsp2).not.toBeNull();
+      rsp2.then((msg) => {
+        expect(msg instanceof GetFileRsp).toBeTruthy();
+        expect(msg.contents).not.toBeUndefined();
+        expect(new TextDecoder('utf-8').decode(new Uint8Array(msg.contents))).toEqual(TEST_STRING + ' ' + TEST_STRING);
+        done();
+      }).catch((ex) => {
+        console.error(ex);
+        fail();
+      });
+    });
+  });
+
+  it('should be able to save the file using PutFileReq when some content is removed', function (done) {
```

Comment:
"should be able to correctly update the file using PutFileReq when some content is removed from the file"

Reply:
updated the description for all cases where offset is 0, <0 and >0.